modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java

```diff
@@ -1954,72 +1960,63 @@ private void _addFileEntry(
 			Map<String, Serializable> values)
 		throws PortalException {
 
-		try {
-			String fileSource = ObjectFieldSettingUtil.getValue(
-				"fileSource", objectField.getObjectFieldSettings());
-
-			if (Objects.equals(fileSource, "documentsAndMedia")) {
-				return;
-			}
+		String fileSource = ObjectFieldSettingUtil.getValue(
+			"fileSource", objectField.getObjectFieldSettings());
 
-			DLFolder dlFileEntryFolder = dlFileEntry.getFolder();
+		if (Objects.equals(fileSource, "documentsAndMedia")) {
+			return;
+		}
 
-			DLFolder dlFolder = _attachmentManager.getDLFolder(
-				dlFileEntry.getCompanyId(), dlFileEntry.getGroupId(),
-				objectField.getObjectFieldId(), serviceContext, userId);
+		DLFolder dlFileEntryFolder = dlFileEntry.getFolder();
 
-			if (Objects.equals(
-					dlFileEntryFolder.getFolderId(), dlFolder.getFolderId())) {
+		DLFolder dlFolder = _attachmentManager.getDLFolder(
+			dlFileEntry.getCompanyId(), dlFileEntry.getGroupId(),
+			objectField.getObjectFieldId(), serviceContext, userId);
 
-				return;
-			}
+		if (Objects.equals(
+				dlFileEntryFolder.getFolderId(), dlFolder.getFolderId())) {
 
-			String originalFileName = TempFileEntryUtil.getOriginalTempFileName(
-				dlFileEntry.getFileName());
+			return;
+		}
 
-			serviceContext.setAttribute(
-				"className", objectDefinition.getClassName());
-			serviceContext.setAttribute("classPK", objectEntryId);
+		String originalFileName = TempFileEntryUtil.getOriginalTempFileName(
+			dlFileEntry.getFileName());
 
-			FileEntry fileEntry = _dlAppLocalService.addFileEntry(
-				null, userId, dlFolder.getRepositoryId(),
-				dlFolder.getFolderId(),
-				DLUtil.getUniqueFileName(
-					dlFileEntry.getGroupId(), dlFolder.getFolderId(),
-					originalFileName, true),
-				dlFileEntry.getMimeType(),
-				DLUtil.getUniqueTitle(
-					dlFileEntry.getGroupId(), dlFolder.getFolderId(),
-					FileUtil.stripExtension(originalFileName)),
-				StringPool.BLANK, null, null, dlFileEntry.getContentStream(),
-				dlFileEntry.getSize(), null, null, null, serviceContext);
+		serviceContext.setAttribute(
+			"className", objectDefinition.getClassName());
+		serviceContext.setAttribute("classPK", objectEntryId);
 
-			if (objectField.isLocalized()) {
-				Map<String, Serializable> localizedValues =
-					(Map<String, Serializable>)values.get(
-						objectField.getI18nObjectFieldName());
+		FileEntry fileEntry = _dlAppLocalService.addFileEntry(
+			null, userId, dlFolder.getRepositoryId(), dlFolder.getFolderId(),
+			DLUtil.getUniqueFileName(
+				dlFileEntry.getGroupId(), dlFolder.getFolderId(),
+				originalFileName, true),
+			dlFileEntry.getMimeType(),
+			DLUtil.getUniqueTitle(
+				dlFileEntry.getGroupId(), dlFolder.getFolderId(),
+				FileUtil.stripExtension(originalFileName)),
+			StringPool.BLANK, null, null, dlFileEntry.getContentStream(),
+			dlFileEntry.getSize(), null, null, null, serviceContext);
 
```

Comment:
@vngoliveira as discussed, I think we should always delete the temporary entry after adding the new file entry based on the temporary file entry.

Reply:
Can you check how the code behaves when using our object layout instead of the form container?